crypto/asn1/a_strex.c

```diff
@@ -601,7 +601,7 @@ int asn1_valid_host(const ASN1_STRING *host)
     const unsigned char *hostptr = host->data;
     int type = host->type;
     int i;
-    char width = -1;
+    signed char width = -1;
```

Comment:
I have found the following:
"If a member of the required source character set enumerated in 2.2.1 is stored in a char object, its value is guaranteed to be positive. If other quantities are stored in a char object, the behavior is implementation-defined: the values are treated as either signed or nonnegative integers." (http://port70.net/~nsz/c/c89/c89-draft.html#3.1.2.5)
"The implementation shall define char to have the same range, representation, and behavior as either signed char or unsigned char." (http://port70.net/~nsz/c/c99/n1256.html#6.2.5)

I guess "unsigned char" being the default is the reason for the above error:
crypto/asn1/a_strex.c:609:15: error: comparison is always false due to limited range of data type [-Werror=type-limits]
if (width == -1 || hostlen == 0)
^~